packages/toc/src/treeview.tsx

```diff
@@ -26,8 +28,18 @@ export class TableOfContentsWidget extends VDomRenderer<TableOfContents.IModel<T
    * includes layout triggered rendering.
    */
   render(): JSX.Element | null {
-    if (!this.model) {
-      return null;
+    if (!this.model || this.model.headings.length === 0) {
+      this.parent?.addClass('jp-TableOfContents-empty');
+      return (
+        <div className="jp-TableOfContents-placeholder">
+          <div className="jp-TableOfContents-placeholderContent">
```

Comment:
Let's make sure to use the JupyterLab theme variables for the type styling: font size, color, etc.